src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverFactoryActivator.java

```diff
@@ -113,6 +114,9 @@
 
     @Reference
     ResourceAccessSecurityTracker resourceAccessSecurityTracker;
+    
+    @Reference(cardinality = ReferenceCardinality.OPTIONAL, policyOption = ReferencePolicyOption.GREEDY)
+    private ResourceResolverMetrics metrics;
```

Comment:
This reference should also be dynamic, this allows that the field is updated when the service appears/disappears instead of restarting the component. The field should be volatile

Reply:
done